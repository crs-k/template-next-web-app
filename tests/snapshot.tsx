import {render} from '@testing-library/react';
import Home from '../src/pages/index';
import {useSession} from 'next-auth/react';
jest.mock('next-auth/react');

it('renders homepage unchanged', () => {
  const mockSession = {
    expires: '100',
    user: {email: 'test@test.com', name: 'Test', image: 'None'}
  };
  (useSession as jest.Mock).mockReturnValueOnce([mockSession, false]);
  const {container} = render(<Home />);
  expect(container).toMatchSnapshot();
});
