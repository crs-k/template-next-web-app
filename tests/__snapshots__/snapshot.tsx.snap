// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center min-h-screen py-2"
  >
    <header>
      <noscript />
      <div
        class="signedInStatus"
      >
        <p
          class="nojs-show loaded"
        >
          <span
            class="notSignedInText"
          >
            You are not signed in
          </span>
          <a
            class="buttonPrimary"
            href="/api/auth/signin"
          >
            Sign in
          </a>
        </p>
      </div>
      <nav>
        <ul
          class="navItems"
        >
          <li
            class="navItem"
          >
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="navItem"
          >
            <a
              href="/client"
            >
              Client
            </a>
          </li>
          <li
            class="navItem"
          >
            <a
              href="/server"
            >
              Server
            </a>
          </li>
          <li
            class="navItem"
          >
            <a
              href="/protected"
            >
              Protected
            </a>
          </li>
          <li
            class="navItem"
          >
            <a
              href="/api-example"
            >
              API
            </a>
          </li>
          <li
            class="navItem"
          >
            <a
              href="/about"
            >
              About
            </a>
          </li>
          <li
            class="navItem"
          >
            <a
              href="/"
            >
              Theme
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main
      class="flex flex-col items-center justify-center w-full flex-1 px-20 text-center"
    >
      <h1
        class="text-6xl font-bold"
      >
        Welcome to
         
        <a
          class="text-blue-600"
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p
        class="mt-3 text-2xl"
      >
        Get started by editing
         
        <code
          class="p-3 font-mono text-lg bg-gray-100 rounded-md"
        >
          pages/index.js
        </code>
      </p>
      <div
        class="flex flex-wrap items-center justify-around max-w-4xl mt-6 sm:w-full"
      >
        <a
          class="p-6 mt-6 text-left border w-96 rounded-xl hover:text-blue-600 focus:text-blue-600"
          href="https://nextjs.org/docs"
        >
          <h3
            class="text-2xl font-bold"
          >
            Documentation →
          </h3>
          <p
            class="mt-4 text-xl"
          >
            Find in-depth information about Next.js features and API.
          </p>
        </a>
        <a
          class="p-6 mt-6 text-left border w-96 rounded-xl hover:text-blue-600 focus:text-blue-600"
          href="https://nextjs.org/learn"
        >
          <h3
            class="text-2xl font-bold"
          >
            Learn →
          </h3>
          <p
            class="mt-4 text-xl"
          >
            Learn about Next.js in an interactive course with quizzes!
          </p>
        </a>
        <a
          class="p-6 mt-6 text-left border w-96 rounded-xl hover:text-blue-600 focus:text-blue-600"
          href="https://github.com/vercel/next.js/tree/master/examples"
        >
          <h3
            class="text-2xl font-bold"
          >
            Examples →
          </h3>
          <p
            class="mt-4 text-xl"
          >
            Discover and deploy boilerplate example Next.js projects.
          </p>
        </a>
        <a
          class="p-6 mt-6 text-left border w-96 rounded-xl hover:text-blue-600 focus:text-blue-600"
          href="https://vercel.com/import?filter=next.js&utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        >
          <h3
            class="text-2xl font-bold"
          >
            Deploy →
          </h3>
          <p
            class="mt-4 text-xl"
          >
            Instantly deploy your Next.js site to a public URL with Vercel.
          </p>
        </a>
      </div>
    </main>
    <footer
      class="footer"
    >
      <hr />
      <ul
        class="navItems"
      >
        <li
          class="navItem"
        >
          <a
            href="https://next-auth.js.org"
          >
            Documentation
          </a>
        </li>
        <li
          class="navItem"
        >
          <a
            href="https://www.npmjs.com/package/next-auth"
          >
            NPM
          </a>
        </li>
        <li
          class="navItem"
        >
          <a
            href="https://github.com/nextauthjs/next-auth-example"
          >
            GitHub
          </a>
        </li>
        <li
          class="navItem"
        >
          <a
            href="/policy"
          >
            Policy
          </a>
        </li>
        <li
          class="navItem"
        >
          <em>
            next-auth@
            4.3.4
          </em>
        </li>
      </ul>
    </footer>
  </div>
</div>
`;
